### DISCUSSION TITLE
Paste from clipboard to a Search Field using Enter Text (wdi5) or SetValue (wdio)

### DISCUSSION THREAD

**Discussion by Scotti-Scholter-Marichal** [ui5-community/wdi5#370] [Q&A]
Good day,

For my application I need to paste some text from my clipboard to a search field. The Uiveri5 equivalent would have been 
`await searchField.sendKeys(protractor.Key.chord(protractor.Key.CONTROL, "v")`
If anyone knows of a wdi5 equivalent I would greatly appreciate it. In the following I will explain what I have already tried.

1. As far as I know wdi5's native API method `enterText` only supports regular text and no key strokes/combinations. Please advise me otherwise.
2. Whenever I try to use `await searchField.setValue(text)` (which is a wdio element method) I get this error :

> Request failed with status 400 due to invalid element state
or
> wdi5 element not interactable: element not interactable

This error occurs regardless of whether the **text** argument is a combination of key strokes or just a regular string. 
I have also checked whether the element exists, is displayed, is enabled and is clickable. All of these are true so I don't know what the problem could be.

I would highly appreciate any help.
Thank you.

Best wishes,
Scotti

**Comment by Siolto** [comment]
Hi @Scotti-Scholter-Marichal,

as far as I am aware of, WebdriverIO does not have a native functionality for pasting values out of you clipboard. Maybe you could try to include the [`node-copy-paste`](https://www.npmjs.com/package/copy-paste) module to your tests. This sounds like it could solve your use case.

Regarding your second issue: Could you maybe create a separate Issue for that and maybe describe it in more detail? Thanks!

Regards
Simon

**Reply by vobu** [reply]
> as far as I am aware of, WebdriverIO does not have a native functionality for pasting values out of you clipboard. Maybe you could try to include the [`node-copy-paste`](https://www.npmjs.com/package/copy-paste) module to your tests. This sounds like it could solve your use case.

hm, it sure sounds it does: https://webdriver.io/docs/api/browser/keys
here's what you could try:
(pidgin-code, attention)
```js
const wdioViaWdi5Input = await browser.asControl(selector).$().$("input") // see https://ui5-community.github.io/wdi5/#/recipes?id=using-wdio-functions
await wdioViaWdi5Input.setValue("") // dummy to bring focus to the <input>
await browser.keys(['Ctrl', 'v']) // send strg+v to the focused <input>
```

**Reply by Scotti-Scholter-Marichal** [reply]
This worked for me! Instead of `.setValue("")` I had to use `.click()` to focus on the input item. Could you ellaborate what the `.$("input")` does in this case? 
Thanks a lot!

**Reply by vobu** [reply]
`browser.asControl(...).$()` transitions from the [wdi5 fluent async api](https://ui5-community.github.io/wdi5/#/usage?id=fluent-async-api) to the [wdio api](https://webdriver.io/docs/api).
The subsequent `$("input")` sub-selects from the previous element, specifically the `<input>` tag
